core/shared/src/test/scala/cats/parse/ParserTest.scala

```diff
@@ -581,6 +592,17 @@ class ParserTest extends munit.ScalaCheckSuite {
     parseTest(Parser.product(fooP, barP), "foobar", ((), ()))
   }
 
+  test("longest match stringIn1") {
+    parseTest(Parser.stringIn1(List("foo", "foobar", "foofoo", "foobat")).string, "foo", "foo")
```

Comment:
can we use a val to not repeat the parser here? It looks like we repeat it 4 times.